handler.go

```diff
@@ -28,18 +29,27 @@ func newHandler(do reflect.Value) *handler {
 }
 
 func newErrorHandler(code int, content string) *handler {
-	h := newHandler(reflect.ValueOf(func() *response.Response {
+	return newHandler(reflect.ValueOf(func() *response.Response {
 		return response.New(content).Status(code)
 	}))
-	return h
 }
 
 func (h *handler) handle(reqURL []string, r *http.Request) *response.Response {
-	ctx, err := h.getUserContexts(reqURL, r)
+	ctx, err := h.fillByCachedUserContexts(h.userContexts, reqURL, r)
 	if err != nil {
 		return response.New(err.Error()).
 			Status(http.StatusBadRequest)
 	}
+
+	if err := h.verify(reqURL, r); err != nil {
+		if err, ok := err.(*VerifyError); ok {
+			return response.New(err.Error()).
+				Status(err.Status())
+		}
+		return response.New(err.Error()).
+			Status(http.StatusBadRequest)
```

Comment:
We already provide the `ValidateError` for return `BadRequest`, so I think this should be changed to `InternalServerError`(500)